packages/editor/src/hooks/block-toolbar.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { __unstableBlockToolbarLastItem as BlockToolbarLastItem } from '@wordpress/block-editor';
+import { ToolbarGroup, ToolbarButton } from '@wordpress/components';
+import { settings as settingsIcon } from '@wordpress/icons';
+import { store as interfaceStore } from '@wordpress/interface';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Override the default edit UI to include a new block inspector control for
+ * assigning a partial syncing controls to supported blocks in the pattern editor.
+ * Currently, only the `core/paragraph` block is supported.
+ *
+ * @param {Component} BlockEdit Original component.
+ *
+ * @return {Component} Wrapped component.
+ */
+const withBlockToolbar = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
+		return (
+			<>
+				<BlockEdit key="edit" { ...props } />
+				{ props.isSelected && <BlockSidebarToggle /> }
+			</>
+		);
```

Comment:
If this ever went to production we'd always need to be careful about hooking into block edit because it can cause performance degradation if not done carefully with the right conditions. In this case the toolbar item will only render for the currently selected block. We may need other conditions.